tests/test_machines.py

```diff
@@ -0,0 +1,377 @@
+"""
```

Comment:
It's using the `unittest` module because these tests were already implemented before.  If you feel strongly I will refactor it to use `pytest`,  but it is working as-is.